control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go

```diff
@@ -256,6 +263,21 @@ func (r *Reconciler) reconcileInPlaceUpgrade(ctx context.Context, nodePoolUpgrad
 	return nil
 }
 
+func maxUnavailable(nodePool *hyperv1.NodePool, nodes []*corev1.Node) (int, error) {
```

Comment:
@yuqi-zhang the unit tests here: https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader_test.go#L271-L388 seem to be already testing for different `maxUnavailable` values (unless I am missing something). Should I add unit tests for this `maxUnavailable()` function similar to https://github.com/openshift/machine-config-operator/blob/f947dbdab6c1380ce5ee4743b5780329ced693d5/pkg/controller/node/node_controller_test.go#L450-L519?

Reply:
Ah right good point. Yes, I think it would be good to add some unit testing for maxUnavailable here as well! Hopefully eventually we can de-duplicate the code and use the MCC directly but for now let's have that additional safety in place.